extensions/hibernate-envers/deployment/src/main/java/io/quarkus/hibernate/envers/deployment/HibernateEnversProcessor.java

```diff
@@ -38,8 +40,14 @@ public void registerEnversReflections(BuildProducer<ReflectiveClassBuildItem> re
         reflectiveClass.produce(
                 new ReflectiveClassBuildItem(false, false, "org.hibernate.tuple.component.DynamicMapComponentTuplizer"));
 
-        buildTimeConfig.revisionListener.ifPresent(s -> reflectiveClass.produce(new ReflectiveClassBuildItem(true, true, s)));
-        buildTimeConfig.auditStrategy.ifPresent(s -> reflectiveClass.produce(new ReflectiveClassBuildItem(true, true, s)));
+        final Map<String, HibernateEnversBuildTimeConfigPersistenceUnit> allPersistenceUnitConfigsAsMap = buildTimeConfig
+                .getAllPersistenceUnitConfigsAsMap();
+        if (allPersistenceUnitConfigsAsMap != null && !allPersistenceUnitConfigsAsMap.isEmpty()) {
+            for (HibernateEnversBuildTimeConfigPersistenceUnit pu : allPersistenceUnitConfigsAsMap.values()) {
+                pu.revisionListener.ifPresent(s -> reflectiveClass.produce(new ReflectiveClassBuildItem(true, true, s)));
+                pu.auditStrategy.ifPresent(s -> reflectiveClass.produce(new ReflectiveClassBuildItem(true, true, s)));
+            }
+        }
```

Comment:
No need to be *that* defensive: `getAllPersistenceUnitConfigsAsMap` never returns null, and if it's empty the loop just won't do anything:

```suggestion
        for (HibernateEnversBuildTimeConfigPersistenceUnit pu : buildTimeConfig.getAllPersistenceUnitConfigsAsMap().values()) {
            pu.revisionListener.ifPresent(s -> reflectiveClass.produce(new ReflectiveClassBuildItem(true, true, s)));
            pu.auditStrategy.ifPresent(s -> reflectiveClass.produce(new ReflectiveClassBuildItem(true, true, s)));
        }
```